sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveUnion.scala

```diff
@@ -182,6 +182,15 @@ object ResolveUnion extends Rule[LogicalPlan] {
             // union have consistent schema.
             aliased += foundAttr
             Alias(addFields(foundAttr, target), foundAttr.name)()
+          case (source: StructType, target: StructType)
```

Comment:
how do we handle different column order for top-level columns?

Reply:
this handled in the code where each of the left attribute it adds its corresponding right side 

```
 val rightProjectList = leftOutputAttrs.map { lattr =>
      val found = rightOutputAttrs.find { rattr => resolver(lattr.name, rattr.name) }
      if (found.isDefined) {
```